src/components/MappingEvents/MappingEventWelcomeDialog.js

```diff
@@ -104,7 +106,12 @@ const UnstyledMappingEventWelcomeDialog = ({ className, mappingEvent, onJoin, on
   const mappingEventWelcomeMessage =
     mappingEvent.welcomeMessage || defaultMappingEventWelcomeMessage;
 
-  const { emailAddress } = getJoinedMappingEventData();
+  const { emailAddress: lastUsedEmailAddress } = getJoinedMappingEventData();
+  let queryEmailAddress: ?string;
+  if (typeof window !== 'undefined') {
+    const queryObject = queryString.parse(window.location.search);
```

Comment:
I'd like to have React Router again at some point…

Reply:
We could do most of that with the next router as well, I think. But it would propagate a bit further, I will take a look, I was not happy with this either.